nano/node/bootstrap/bootstrap.cpp

```diff
@@ -573,6 +592,158 @@ void nano::bootstrap_attempt::add_bulk_push_target (nano::block_hash const & hea
 	bulk_push_targets.emplace_back (head, end);
 }
 
+void nano::bootstrap_attempt::attempt_restart_check (nano::unique_lock<std::mutex> & lock_a)
+{
+	if (!confirmed_frontiers && requeued_pulls > (!node->network_params.network.is_test_network () ? nano::bootstrap_limits::requeued_pulls_limit : nano::bootstrap_limits::requeued_pulls_limit_test))
+	{
+		confirm_frontiers (lock_a);
+		if (!confirmed_frontiers)
+		{
+			node->stats.inc (nano::stat::type::bootstrap, nano::stat::detail::frontier_confirmation_failed, nano::stat::dir::in);
+			node->bootstrap_initiator.blacklist.add (endpoint_frontier_request);
```

Comment:
Would be nice if the blacklisting (with IP/port) was always_log`ed.

Reply:
Done